src/linear_algebra/clifford_algebra.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2020 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser, Utensil Song.
+-/
+
+import algebra.ring_quot
+import linear_algebra.tensor_algebra
+import linear_algebra.quadratic_form
+
+/-!
+# Clifford Algebras
+
+We construct the Clifford algebra of a module `M` over a commutative ring `R`, equipped with
+a quadratic_form `Q`.
+
+## Notation
+
+The Clifford algebra of the `R`-module `M` equipped with a quadratic_form `Q` is denoted as
+`clifford_algebra R M Q`.
+
+Given a linear morphism `f : M → A` from a semimodule `M` to another `R`-algebra `A`, such that
+`cond : ∀ m, f m * f m = algebra_map _ _ (Q m)`, there is a (unique) lift of `f` to an `R`-algebra
+morphism, which is denoted `clifford_algebra.lift R Q f cond`.
+
+The canonical linear map `M → clifford_algebra R M Q` is denoted `clifford_algebra.ι R Q`.
+
+## Theorems
+
+The main theorems proved ensure that `clifford_algebra R M Q` satisfies the universal property
+of the clifford algebra.
+1. `ι_comp_lift` is the fact that the composition of `ι R Q` with `lift R Q f cond` agrees with `f`.
+2. `lift_unique` ensures the uniqueness of `lift R Q f cond` with respect to 1.
+
+## Implementation details
+
+The clifford algebra of `M` is constructed as a quotient of the tensor algebra, as follows.
+1. We define a relation `clifford_algebra.rel R M Q` on `tensor_algebra R M`.
+   This is the smallest relation which identifies squares of elements of `M` with `Q m`.
+2. The clifford algebra is the quotient of the tensor algebra by this relation.
+
+This file is almost identical to `linear_algebra/exterior_algebra.lean`.
+Indeed, the exterior algebra can be recovered by setting `Q` to `0`.
+-/
+
+variables (R : Type*) [comm_ring R]
+variables (M : Type*) [add_comm_group M] [module R M]
+variables (Q : quadratic_form R M)
+
+variable {n : ℕ}
+
+namespace clifford_algebra
+open tensor_algebra
+
+/-- `rel` relates each `ι m * ι m`, for `m : M`, with `Q m`.
+
+The clifford algebra of `M` is defined as the quotient modulo this relation.
+-/
+inductive rel : tensor_algebra R M → tensor_algebra R M → Prop
+| of (m : M) : rel (ι R m * ι R m) (algebra_map R _ (Q m))
+
+end clifford_algebra
+
+/--
+The Clifford algebra of an `R`-module `M` equipped with a quadratic_form `Q`.
+-/
+@[derive [inhabited, semiring, algebra R]]
+def clifford_algebra := ring_quot (clifford_algebra.rel R M Q)
+
+namespace clifford_algebra
+
+variables {M}
+
+/-
+The canonical linear map `M →ₗ[R] clifford_algebra R M Q`.
+-/
+def ι : M →ₗ[R] clifford_algebra R M Q :=
+(ring_quot.mk_alg_hom R _).to_linear_map.comp (tensor_algebra.ι R)
+
+variables {R}
+
+/-- As well as being linear, `ι m` squares to the quadratic form -/
+@[simp]
+theorem ι_square_scalar (m : M) : ι R Q m * ι R Q m = algebra_map R _ (Q m) :=
+begin
+  dsimp [ι],
+  rw ←alg_hom.map_mul,
+  rw ring_quot.mk_alg_hom_rel R (rel.of m),
+  rw alg_hom.commutes,
```

Comment:
```suggestion
  erw [←alg_hom.map_mul, ring_quot.mk_alg_hom_rel R (rel.of m), alg_hom.commutes],
```
Using `erw` you don't need to unfold `\iota`. You can also combine the three rewrites into one line.

Reply:
I've tried to make these proofs resemble the ones in `exterior_algebra`. Perhaps I'll apply your suggestions to both.